Composer/packages/client/src/pages/botProject/GetAppInfoFromPublishProfileDialog.tsx

```diff
@@ -0,0 +1,125 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT License.
+
+/** @jsx jsx */
+import React, { Fragment, useState, useEffect } from 'react';
+import { jsx } from '@emotion/core';
+import { useRecoilValue } from 'recoil';
+import { PublishTarget } from '@bfc/shared';
+import formatMessage from 'format-message';
+import { DefaultButton, PrimaryButton } from 'office-ui-fabric-react/lib/Button';
+import Dialog, { DialogFooter } from 'office-ui-fabric-react/lib/Dialog';
+import { Dropdown, IDropdownOption } from 'office-ui-fabric-react/lib/Dropdown';
+
+import { settingsState } from '../../recoilModel/atoms/botState';
+
+type AppInfo = {
+  appId: string;
+  appPassword?: string;
+};
+
+type Props = {
+  projectId: string;
+  hidden?: boolean;
+  onOK: (info: AppInfo) => void;
+  onCancel: () => void;
+};
+
+export const GetAppInfoFromPublishProfileDialog: React.FC<Props> = (props) => {
+  const { projectId, hidden, onOK: onAdd, onCancel } = props;
+  const { publishTargets } = useRecoilValue(settingsState(projectId));
+  const [publishTargetOptions, setPublishTargetOptions] = useState<IDropdownOption[]>([]);
+  const [publishTargetsErrorMessage, setPublishTargetsErrorMessage] = useState<string | undefined>();
+  const [selectedKey, setSelectedKey] = useState<string | number | undefined>();
+
+  const dialogTitle = {
+    title: formatMessage('Add from publishing profile'),
+    subText: formatMessage('Select the publishing profile you’d like to add a Microsoft App ID and Password from.'),
+  };
+
+  const getAppInfo = (profile: PublishTarget) => {
+    if (profile) {
+      const config = JSON.parse(profile.configuration);
+      const appId = config?.settings?.MicrosoftAppId;
+      const appPassword = config?.settings?.MicrosoftAppPassword;
+
+      if (appId) {
```

Comment:
why this check on appId only

Reply:
It may be that with the new integration security model that the password is empty in the future.  If it has an app ID that is enough to have it in the list.